pkg/storage/unified/resource/server_test.go

```diff
@@ -234,3 +235,285 @@ func TestSimpleServer(t *testing.T) {
+
+	testCases := []struct {
+		name           string
+		initialFolder  string
+		targetFolder   string
+		permissionMap  map[string]bool
+		expectSuccess  bool
+		expectedChecks int
+		expectedError  int32 // HTTP status code for error, 0 if no error expected
+	}{
+		{
+			name:           "Create resource in folder",
+			initialFolder:  "folder1",
+			targetFolder:   "", // No move, just create
+			permissionMap:  map[string]bool{"folder1:create": true},
+			expectSuccess:  true,
+			expectedChecks: 1,
+			expectedError:  0,
+		},
+		{
+			name:           "Create resource denied",
+			initialFolder:  "folder1",
+			targetFolder:   "", // No move, just create
+			permissionMap:  map[string]bool{"folder1:create": false},
+			expectSuccess:  false,
+			expectedChecks: 1,
+			expectedError:  http.StatusForbidden,
+		},
+		{
+			name:           "Update resource in same folder",
+			initialFolder:  "folder1",
+			targetFolder:   "folder1", // Same folder, just update
+			permissionMap:  map[string]bool{"folder1:update": true},
+			expectSuccess:  true,
+			expectedChecks: 1,
+			expectedError:  0,
+		},
+		{
+			name:           "Update resource denied",
+			initialFolder:  "folder1",
+			targetFolder:   "folder1", // Same folder
+			permissionMap:  map[string]bool{"folder1:update": false},
+			expectSuccess:  false,
+			expectedChecks: 1,
+			expectedError:  http.StatusForbidden,
+		},
+		{
+			name:           "Move resource to another folder - allowed",
+			initialFolder:  "folder1",
+			targetFolder:   "folder2", // Moving to a different folder
+			permissionMap:  map[string]bool{"folder1:update": true, "folder2:create": true},
+			expectSuccess:  true,
+			expectedChecks: 2, // Should check both folders
+			expectedError:  0,
+		},
+		{
+			name:           "Move resource - source folder access denied",
+			initialFolder:  "folder1",
+			targetFolder:   "folder2",
+			permissionMap:  map[string]bool{"folder1:update": false, "folder2:create": true},
+			expectSuccess:  false,
+			expectedChecks: 1, // Should stop at first check
+			expectedError:  http.StatusForbidden,
+		},
+		{
+			name:           "Move resource - destination folder access denied",
+			initialFolder:  "folder1",
+			targetFolder:   "folder2",
+			permissionMap:  map[string]bool{"folder1:update": true, "folder2:create": false},
+			expectSuccess:  false,
+			expectedChecks: 2, // Should do both checks but fail on second
+			expectedError:  http.StatusForbidden,
+		},
+		{
+			name:           "Move resource - from empty folder to named folder",
+			initialFolder:  "",
+			targetFolder:   "folder1",
+			permissionMap:  map[string]bool{":update": true, "folder1:create": true},
+			expectSuccess:  true,
+			expectedChecks: 2,
+			expectedError:  0,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Reset the backend for each test
+			store, err := NewCDKBackend(context.Background(), CDKBackendOptions{
+				Bucket: memblob.OpenBucket(nil),
+			})
```

Comment:
TODO: I don't think we want to keep the CDK Backend. Let's call this test with the sql backend.